synedrion/src/www02/bip32.rs

```diff
@@ -0,0 +1,206 @@
+use core::fmt::Debug;
+
+use crate::{
+    curve::{Point, Scalar},
+    tools::Secret,
+    ProductionParams112, SchemeParams, TestParams,
+};
+use alloc::vec::Vec;
+use bip32::{ChainCode, DerivationPath, PrivateKey as _, PrivateKeyBytes};
+
+use digest::typenum::Unsigned;
+use digest::Digest;
+use ecdsa::{hazmat::DigestPrimitive, SigningKey, VerifyingKey};
+use primeorder::elliptic_curve::{
+    sec1::{FromEncodedPoint, ModulusSize, ToEncodedPoint},
+    Curve, CurveArithmetic, PrimeCurve, PublicKey, SecretKey,
+};
+use serde::{Deserialize, Serialize};
+use tiny_curve::{PrivateKeyBip32, PublicKeyBip32};
+
+use super::ThresholdKeyShare;
+
+/// Used for deriving child keys from a parent type.
+pub trait DeriveChildKey<C: CurveArithmetic + PrimeCurve>: Sized {
+    /// Return a verifying key derived from the given type using the BIP-32 scheme.
+    fn derive_verifying_key_bip32(&self, derivation_path: &DerivationPath) -> Result<VerifyingKey<C>, bip32::Error>;
+}
+
+mod sealed {
```

Comment:
What's up with sealing? Do these traits show up in the docs otherwise?

Reply:
The idea was to lock down the types for which it's ok to impl them, not really about hiding from docs. You think it's overkill? They have to be pub because the types they are impl'd for are pub.